semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/package.scala

```diff
@@ -4,76 +4,82 @@ import gremlin.scala.GremlinScala
-    def source: nodes.Expression = call.argument(2)
+  class AssignmentExt(val assignment: Assignment) extends AnyVal {
+    def target: nodes.Expression = assignment.argument(1)
+    def source: nodes.Expression = assignment.argument(2)
   }
 
-  class AssignmentTrav(val wrapped: Steps[nodes.Call]) extends AnyVal {
-    private def raw: GremlinScala[nodes.Call] = wrapped.raw
+  class AssignmentTrav(val wrapped: Steps[Assignment]) extends AnyVal {
+    private def raw: GremlinScala[Assignment] = wrapped.raw
     def target: NodeSteps[nodes.Expression] = wrapped.map(_.target)
     def source: NodeSteps[nodes.Expression] = wrapped.map(_.source)
   }
 
   class OpAstNodeExt[A <: nodes.AstNode](val node: A) extends AnyVal {
-    def inAssignment: NodeSteps[nodes.Call] =
-      node.start.inAstMinusLeaf.isCall.name(NodeTypeStarters.assignmentPattern)
+    def inAssignment: NodeSteps[Assignment] =
+      node.start.inAstMinusLeaf.isCall.name(NodeTypeStarters.assignmentPattern).map(new Assignment(_))
 
-    def assignments: NodeSteps[nodes.Call] =
-      rawTravForPattern(NodeTypeStarters.assignmentPattern)
+    def assignments: NodeSteps[Assignment] =
+      rawTravForPattern(NodeTypeStarters.assignmentPattern).map(new Assignment(_))
 
-    def arithmetics: NodeSteps[nodes.Call] =
-      rawTravForPattern(NodeTypeStarters.arithmeticPattern)
+    def arithmetics: NodeSteps[Arithmetic] =
+      rawTravForPattern(NodeTypeStarters.arithmeticPattern).map(new Arithmetic(_))
 
     private def rawTravForPattern(pattern: String): NodeSteps[nodes.Call] =
       node.ast.isCall.name(pattern)
   }
 
   class OpAstNodeTrav[A <: nodes.AstNode](val wrapped: Steps[A]) extends AnyVal {
     private def raw: GremlinScala[A] = wrapped.raw
-    def inAssignment: NodeSteps[nodes.Call] = wrapped.flatMap(_.inAssignment)
-    def assignments: NodeSteps[nodes.Call] = wrapped.flatMap(_.assignments)
-    def arithmetics: NodeSteps[nodes.Call] = wrapped.flatMap(_.arithmetics)
+    def inAssignment: NodeSteps[Assignment] = wrapped.flatMap(_.inAssignment)
+    def assignments: NodeSteps[Assignment] = wrapped.flatMap(_.assignments)
+    def arithmetics: NodeSteps[Arithmetic] = wrapped.flatMap(_.arithmetics)
   }
 
   class TargetExt(val expr: nodes.Expression) extends AnyVal {
-    def isArrayAccess: NodeSteps[nodes.Call] =
+    def isArrayAccess: NodeSteps[ArrayAccess] =
       expr.ast.isCall
         .nameExact(Operators.computedMemberAccess,
                    Operators.indirectComputedMemberAccess,
                    Operators.indexAccess,
                    Operators.indirectIndexAccess)
+        .map(new ArrayAccess(_))
   }
 
   class TargetTrav(val wrapped: NodeSteps[nodes.Expression]) extends AnyVal {
     private def raw: GremlinScala[nodes.Expression] = wrapped.raw
-    def isArrayAccess: NodeSteps[nodes.Call] = wrapped.flatMap(_.isArrayAccess)
+    def isArrayAccess: NodeSteps[ArrayAccess] = wrapped.flatMap(_.isArrayAccess)
```

Comment:
Same as for OpAstNodeTrav. The hole class is unnecessary and the methods should go to Expression.scala

Reply:
No, as explained above. This is about augmenting `Expression`.

Reply:
Same as above.